vhdbuilder/packer/test/run-test.sh

```diff
@@ -15,15 +15,48 @@ trap cleanup EXIT
 
 DISK_NAME="${TEST_RESOURCE_PREFIX}-disk"
 VM_NAME="${TEST_RESOURCE_PREFIX}-vm"
-az disk create --resource-group $RESOURCE_GROUP_NAME \
-    --name $DISK_NAME \
-    --source "${OS_DISK_URI}"  \
-    --query id
-az vm create --name $VM_NAME \
+
+if [ "$MODE" == "sigMode" ]; then
+	echo "SIG existence checking for $MODE"
+	id=$(az sig show --resource-group ${AZURE_RESOURCE_GROUP_NAME} --gallery-name ${SIG_GALLERY_NAME}) || id=""
+	if [ -z "$id" ]; then
+		echo "Shared Image gallery ${SIG_GALLERY_NAME} does not exist in the resource group ${AZURE_RESOURCE_GROUP_NAME} location ${AZURE_LOCATION}"
+        exit 1
+	fi
+
+	id=$(az sig image-definition show \
+		--resource-group ${AZURE_RESOURCE_GROUP_NAME} \
+		--gallery-name ${SIG_GALLERY_NAME} \
+		--gallery-image-definition ${SIG_IMAGE_NAME}) || id=""
+	if [ -z "$id" ]; then
+		echo "Creating image definition ${SIG_IMAGE_NAME} does not exist in gallery ${SIG_GALLERY_NAME} resource group ${AZURE_RESOURCE_GROUP_NAME}"
+        exit 1
+	fi
+
+    IMG_DEF="/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/galleries/${SIG_GALLERY_NAME}/images/${SIG_IMAGE_NAME}/versions/${SIG_IMAGE_VERSION}"
+
+    # In SIG mode, Windows VM requires admin-username and admin-password to be set,
+    # otherwise 'root' is used by default but not allowed by the Windows Image. See the error image below:
+    # ERROR: This user name 'root' meets the general requirements, but is specifically disallowed for this image. Please try a different value.
+    az vm create\
     --resource-group $RESOURCE_GROUP_NAME \
-    --attach-os-disk $DISK_NAME \
-    --os-type $OS_TYPE  \
+    --name $VM_NAME \
+    --image $IMG_DEF \
+    --admin-username azureuser \
+    --admin-password "TestVM@$(date +%s)" \
```

Comment:
Do we need to log password for debug/trouble shooting/check?

Reply:
username and password is required for SIG mode, but when SIG is not enabled, we cannot specify the password.  We can introduce a debug toggle, when enabled we can keep the vm for debugging.